server/filestore.go

```diff
@@ -186,42 +188,44 @@ type fileStore struct {
 // Represents a message store block and its data.
 type msgBlock struct {
 	// Here for 32bit systems and atomic.
-	first   msgId
-	last    msgId
-	mu      sync.RWMutex
-	fs      *fileStore
-	aek     cipher.AEAD
-	bek     cipher.Stream
-	seed    []byte
-	nonce   []byte
-	mfn     string
-	mfd     *os.File
-	cmp     StoreCompression // Effective compression at the time of loading the block
-	liwsz   int64
-	index   uint32
-	bytes   uint64 // User visible bytes count.
-	rbytes  uint64 // Total bytes (raw) including deleted. Used for rolling to new blk.
-	msgs    uint64 // User visible message count.
-	fss     map[string]*SimpleState
-	kfn     string
-	lwts    int64
-	llts    int64
-	lrts    int64
-	llseq   uint64
-	hh      hash.Hash64
-	cache   *cache
-	cloads  uint64
-	cexp    time.Duration
-	ctmr    *time.Timer
-	werr    error
-	dmap    avl.SequenceSet
-	fch     chan struct{}
-	qch     chan struct{}
-	lchk    [8]byte
-	loading bool
-	flusher bool
-	noTrack bool
-	closed  bool
+	first      msgId
+	last       msgId
+	mu         sync.RWMutex
+	fs         *fileStore
+	aek        cipher.AEAD
+	bek        cipher.Stream
+	seed       []byte
+	nonce      []byte
+	mfn        string
+	mfd        *os.File
+	cmp        StoreCompression // Effective compression at the time of loading the block
+	liwsz      int64
+	index      uint32
+	bytes      uint64 // User visible bytes count.
+	rbytes     uint64 // Total bytes (raw) including deleted. Used for rolling to new blk.
+	msgs       uint64 // User visible message count.
+	fss        map[string]*SimpleState
+	kfn        string
+	lwts       int64
+	llts       int64
+	lrts       int64
+	llseq      uint64
+	hh         hash.Hash64
+	cache      *cache
+	cloads     uint64
+	cexp       time.Duration
+	ctmr       *time.Timer
+	werr       error
+	dmap       avl.SequenceSet
+	fch        chan struct{}
+	qch        chan struct{}
+	lchk       [8]byte
+	loading    bool
+	flusher    bool
+	noTrack    bool
+	needSync   bool
```

Comment:
AFAICT `needSync` is never not `false`?

Reply:
Thanks good catch, was getting late..

Fixed in separate CL but once LGTM I will squash.